Topic-4-JSDP/classes.js

```diff
@@ -0,0 +1,74 @@
+class EventEmitter {
+    constructor() {
+        this.events = {};
+    };
+    on (eventName, callback){
+        if (!this.events[eventName]) {
+            this.events[eventName] = [];
+        }
+        this.events[eventName].push(callback);
+    };
+    emit (eventName) {
+        let event = this.events[eventName];
+        event.forEach(callback => {
+            callback.call();
+        });
+    };
+    off (eventName, callback) {
+        let event = this.events[eventName];
+        if (!event) return this;
+        for (let i = event.length; i >= 0; i--) {
+            if (event[i] === callback) {
+                event.splice(i, 1);
+                break;
+            }
+        }
+    };
+}
+
+class Movie extends EventEmitter {
+    constructor(name, year, duration){
+        super();
+        this.title = name;
+        this.year = year;
+        this.duration = duration;
+    }
+    play(){
+        newLogger.log("The '" + Movie.prototype.play.name + "' event has been emitted");
```

Comment:
Check the usage of `this` instead of the prototype. It will be like a super power once you understand it, ahah. Same for the following methods (pause and resume)